sdk/entra/WebJobs.Extensions.AuthenticationEvents/src/WebJobs.Extensions.AuthenticationEvents/AuthEventResponseHandler.cs

```diff
@@ -0,0 +1,230 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Buffers;
+using System.Globalization;
+using System.IO;
+using System.Net.Http;
+using System.Text;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+
+using Microsoft.AspNetCore.Http;
+using Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents.Framework;
+using Microsoft.Azure.WebJobs.Host.Bindings;
+
+namespace Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents
+{
+    /// <summary>Contains the IActionRequest response when the function is called.</summary>
+    /// <seealso cref="IValueBinder" />
+    public class AuthEventResponseHandler : IValueBinder
+    {
+        /// <summary>The response property.</summary>
+        internal const string EventResponseProperty = "$event$response";
+
+        /// <summary>Gets or sets the action result.</summary>
+        /// <value>The action result.</value>
+        public AuthEventResponse Response { get;  internal set; }
+
+        /// <summary>Gets the type.</summary>
+        /// <value>The type.</value>
+        public Type Type => typeof(AuthEventResponse).MakeByRefType();
+
+        /// <summary>Gets or sets the request.</summary>
+        /// <value>The associated request.</value>
+        public AuthEventRequestBase Request { get;  internal set; }
+
+        /// <summary>Gets the value asynchronous.</summary>
+        /// <returns>
+        ///   <br />
+        /// </returns>
+        public Task<object> GetValueAsync()
+        {
+            return Task.FromResult<object>(Response);
+        }
+
+        /// <summary>Sets the value asynchronous.</summary>
+        /// <param name="result">The result.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns>
+        ///   A task flag as completed.
+        /// </returns>
+        public Task SetValueAsync(object result, CancellationToken cancellationToken)
+        {
+            try
+            {
+                if (result == null)
+                    throw new InvalidCastException(AuthEventResource.Ex_Invalid_Return);
```

Comment:
Why not ArgumentNullException?

Reply:
My thinking is that the webjobs framework was handling the cast and if I got a null, it was because the cast failed, but thinking about it ArgNull is way better. Thanks, made the change.